crates/scheduler/src/call/crypto/hash.rs

```diff
@@ -61,15 +65,30 @@ fn initialize<B: Board>(mut call: SchedulerCall<B, api::initialize::Sig>) {
     let api::initialize::Params { algorithm } = call.read();
     let scheduler = call.scheduler();
     let result = try {
-        let context = match convert_hash_algorithm::<B>(*algorithm)?? {
+        let context_or_error = match convert_hash_algorithm::<B>(*algorithm)?? {
             #[cfg(feature = "board-api-crypto-sha256")]
-            Algorithm::Sha256 => HashContext::Sha256(board::crypto::Sha256::<B>::default()),
+            Algorithm::Sha256 => {
+                let hash_or_error = HashApi::<board::crypto::Sha256<B>>::new();
+                match hash_or_error {
+                    Ok(hash) => Ok(HashContext::Sha256(hash)),
+                    Err(error) => Err(error),
+                }
```

Comment:
I didn't think of returning the result directly, this is actually much better.

nit: you can actually improve the style a bit here
```suggestion
                HashApi::<board::crypto::Sha256<B>>::new().map(HashContext::Sha256)
```

Reply:
Nice, and I had to make a modification -- `HashContext::Sha256()` does not take `HashApi` but the hash in it.